packit/base_git.py

```diff
@@ -182,14 +191,15 @@ def run_action(self, action: ActionName, method: Callable = None, *args, **kwarg
 
         :param action: ActionName enum (Name of the action that can be overwritten
                                                 in the package_config.actions)
+        :param openshift_deployer: OpenshiftDeployer to execute actions in a POD.
         :param method: method to run if the action was not defined by user
                     (if not specified, the action can be used for custom hooks)
         :param args: args for the method
         :param kwargs: kwargs for the method
         """
         if not method:
             logger.debug(f"Running {action} hook.")
-        if self.with_action(action=action):
+        if self.with_action(action=action, openshift_deployer=openshift_deployer):
```

Comment:
I'd say that easier and cleaner design would be to pass something to the BaseGit constructor, not pass things to every method. I'd also pass RunCommandHandler.

I'm also thinking what's the best way to handle the relationship b/w sandcastle and the openshift handler. But since the Deployer class needs configuration (image, namespace) it feels like a good design what you propose: initialize the class in service and pass the instance here.

Reply:
Hopefully, solved properly. Constructor without type like `sandcastle: OpenshiftDeployer = None` solves a lot of trouble. Like `from generator.deploy....`

I have committed the changes in small commits.